src/components/SearchOverlay.vue

```diff
@@ -117,16 +120,50 @@ async function search() {
     }
     const variationName = breakpointService.active.value.toUpperCase();
 
+    const selectedCategoryFilterIds = filters.value['category'];
+    const categoryFilterThreshold = contextStore.context.value.allowThirdLevelCategories ? 3 : 2;
+
     const request = new SearchCollectionBuilder()
         .addRequest(new ProductSearchBuilder(contextStore.defaultSettings)
             .setSelectedProductProperties(contextStore.selectedProductProperties)
             .setSelectedVariantProperties({ allData: true })
             .setTerm(filters.value.term.length > 0 ? filters.value.term : null)
-            .facets(f => f
-                .addCategoryFacet('ImmediateParent', Array.isArray(filters.value['category']) && filters.value['category']?.length > 0 ? filters.value['category'] : null)
-                .addBrandFacet(Array.isArray(filters.value['brand']) && filters.value['brand']?.length > 0 ? filters.value['brand'] : null)
-                .addSalesPriceRangeFacet('Product', applySalesPriceFacet ? Number(filters.value.price[0]) : undefined, applySalesPriceFacet ? Number(filters.value.price[1]) : undefined),
-            )
+            .filters(f => {
+                if (Array.isArray(selectedCategoryFilterIds)) {
+                    selectedCategoryFilterIds.slice(0, (contextStore.context.value.allowThirdLevelCategories ? 3 : 2)).forEach(id => {
+                        f.addProductCategoryIdFilter('Ancestor', id);
+                    });
+                }
+            })
+            .facets(f => {
+
+                let forFacet = undefined;
+                if (Array.isArray(selectedCategoryFilterIds) && selectedCategoryFilterIds.length > 0) {
+                    if (selectedCategoryFilterIds.length < categoryFilterThreshold) {
+                        forFacet = [{
+                            breadcrumbPathStartingFromRoot: selectedCategoryFilterIds.map(id => ({ id } as CategoryNameAndId)),
```

Comment:
Behøver du caste her?
Kan du ikke angive typen på din variable i stedet for, for at få den til at være glad :)

Reply:
check